File: Exercises/C++/Dietel & Dietel C++ How to Program/Chapter_2/Exercises/Exercise2_19_Smallest&Largest/SmallestLargest/SmallestLargest/SmallestLargest.cpp
// SmallestLargest.cpp : This file contains the 'main' function. Program execution begins and ends there.
//

#include <iostream>

using namespace std;

int main()
{
    int number1;
    int number2;
    int number3;
    int sum;
    int average;
    int product;

    cout << "Enter three different integers: ";
    cin >> number1 >> number2 >> number3;

    sum = number1 + number2 + number3;
    cout << "Sum is " << sum << endl;
    
    average = (number1 + number2 + number3) / 3;
    cout << "Average is " << average << endl;
    
    product = number1 * number2 * number3;
    cout << "Product is " << product << endl;

    int min = number1;
    if (min > number2)
        min = number2;
    if (min > number3)
        min = number3;
    cout << "Smallest is " << min << endl;

    int max = number1;
    if (max < number2) 
        max = number2;
    if (max < number3) 
        max = number3;
    cout << "Largest is " << max << endl;

    system("pause");
}

// Run program: Ctrl + F5 or Debug > Start Without Debugging menu
// Debug program: F5 or Debug > Start Debugging menu

// Tips for Getting Started: 
//   1. Use the Solution Explorer window to add/manage files
//   2. Use the Team Explorer window to connect to source control
//   3. Use the Output window to see build output and other messages
//   4. Use the Error List window to view errors
//   5. Go to Project > Add New Item to create new code files, or Project > Add Existing Item to add existing code files to the project
//   6. In the future, to open this project again, go to File > Open > Project and select the .sln file

File: Exercises/Java/Java_How to Program_8th Edition/Visual Studio/Chapter 2_Introduction to Java Applications/Exercises/SmallestLargest/SmallestLargest/src/main/java/pkg/SmallestLargest.java
package pkg;

import java.util.Scanner;

public class SmallestLargest 
{
	public static void main(String[] args) 
	{
		int number1;
		int number2;
		int number3;
		int sum;
		int average;
		int product;
		int largest;
		int smallest;

		Scanner input = new Scanner(System.in);

		System.out.print("Enter first integer: ");
		number1 = input.nextInt();

		System.out.print("Enter second integer: ");
		number2 = input.nextInt();

		System.out.print("Enter third integer: ");
		number3 = input.nextInt();

		sum = number1 + number2 + number3;
		average = (number1 + number2 + number3) / 3;
		product = number1 * number2 * number3;

		System.out.printf("Sum is %d\n", sum);
		System.out.printf("Average is %d\n", average);
		System.out.printf("Product is %d\n", product);

		largest = number1; // assume number1 is the largest to start with
		smallest = number1; // assume number1 is the smallest to start with

		if (number2 > largest) 
			largest = number2;

		if (number3 > largest) 
			largest = number3;

		if (number2 < smallest)
			smallest = number2;

		if (number3 < smallest)
			smallest = number3;

		System.out.printf("Smallest number is %d\n", smallest);
		System.out.printf("Largest number is %d\n", largest);
	}
}
